ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala

```diff
@@ -480,11 +481,27 @@ object BigDiffy extends Command with Serializable {
     (r: GenericRecord) =>  MultiKey(xs.map(x => get(x, 0, r)))
   }
 
-  private[diffy] def tableRowKeyFn(keys: Seq[String]): TableRow => MultiKey = {
+  private[diffy] def tableRowKeyFn(keys: Seq[String], convertNulls: Boolean = false): TableRow => MultiKey = {
     @tailrec
     def get(xs: Array[String], i: Int, r: java.util.Map[String, AnyRef]): String =
       if (i == xs.length - 1) {
-        r.get(xs(i)).toString
+        // Convert nulls to "null" if convertNulls specified
+        // This makes it easier to compare final datasets where null
+        // may be a valid string category for example
+        // Note this can also be done via Java's String.valueOf(_)
+
+        val keyValue: Option[String] = Option(r.get(xs(i))).map(_.toString)
+
+        (keyValue, convertNulls) match {
+          case (Some(value), _) => value
+          case (None, true) => "null"
+          case (None, false) => {
+          logger.error(
+            s"""Null value found for key: ${xs.mkString(".")}.
```

Comment:
Should match the existing behaviour with Avro. https://github.com/spotify/ratatool/blob/master/ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala#L465

Reply:
Probably can reuse most of the existing avro code

Reply:
Do you also want to warn here? I was just thinking in our case it was like a valid category, so we had hundreds of thousands of rows like that (thus wanting to make it a command line argument before).

Reply:
Yea I think it makes sense to warn here as well, I'd rather err on the side of being too noisy first and course correct later if it's a problem